core/controller/src/main/scala/whisk/core/controller/actions/SequenceActions.scala

```diff
@@ -280,6 +276,9 @@ protected[actions] trait SequenceActions {
         // since all throwables are recovered with a failed accounting instance and this is
         // in turned boxed to FailedSequenceActivation
         case FailedSequenceActivation(accounting) => Future.successful(accounting)
+        case _: NoDocumentException =>
+          Future.successful(
+            initialAccounting.fail(ActivationResponse.applicationError(sequenceComponentNotFound), None))
```

Comment:
at a quick glance i dont think this is quite right - it's possible there were components in the sequence executed before you reach the point of failure due to a missing action in the sequence. Using the initial accounting object here throws away the intermediate record up to that point in this patch.

I suggest a recover with after `resolvedFutureActions` and before `foldLeft` otherwise convince me this is right with a test :)

Reply:
I'm wrong - looking deeper:

```
val resolvedFutureActions = resolveDefaultNamespace(components, user) map { c =>
      WhiskActionMetaData.resolveActionAndMergeParameters(entityStore, c)
    }
```

Will resolve all the components first and so if the future fails with a no doc exception, it is safe to use the initial accounting object. The placement of the recover with is dangerous though (for the reason I initially thought it was incorrect).